File: javaexample/src/main/java/com/cachesmith/example/java/dao/TesteDataSource.java
package com.cachesmith.example.java.dao;

import android.database.sqlite.SQLiteOpenHelper;

import com.cachesmith.library.DataSource;
import com.cachesmith.library.annotations.Entity;

@Entity("com.cachesmith.example.models.Teste")
public class TesteDataSource extends DataSource {
    public TesteDataSource(SQLiteOpenHelper dbHelper) {
        super(dbHelper);
    }
}

File: example/src/main/java/com/cachesmith/example/dao/TesteDataSource.kt
package com.cachesmith.example.dao

import android.database.sqlite.SQLiteOpenHelper
import com.cachesmith.example.models.Teste
import com.cachesmith.library.DataSource
import com.cachesmith.library.annotations.Entity

@Entity("com.cachesmith.example.models.Teste")
class TesteDataSource(dbHelper: SQLiteOpenHelper) : DataSource(dbHelper) {
    fun insert(teste: Teste) {
        database.execSQL("")
    }
}